### PR TITLE
[rv_dm] Implement bad address error on SBA

### PR DESCRIPTION
**Pull request by msfschaffner**
This addresses https://github.com/lowRISC/opentitan/issues/11480.

The fix has been PR'ed upstream as well (#129), but until that is merged, the patch is hosted on the lowrisc patch repository so that we can vendor it in.

@sriyerg I believe the DV environment now needs to be updated to correctly predict this error.
Let me know if I should do that as part of this PR.

### CODE REVIEW

**Comment by sriyerg** [comment]
> @sriyerg I believe the DV environment now needs to be updated to correctly predict this error. Let me know if I should do that as part of this PR.

Yes - this will break some tests, but they are designated V2. The vendor-in should not break our CI so feel free to leave that to me.

**Comment by msfschaffner** [comment]
> > @sriyerg I believe the DV environment now needs to be updated to correctly predict this error. Let me know if I should do that as part of this PR.
> 
> Yes - this will break some tests, but they are designated V2. The vendor-in should not break our CI so feel free to leave that to me.

Sounds good!

I've pushed an updated version where the bus-integrity error is hooked up to the `other_err` input that we've just added to the `rv_dm` (see https://github.com/pulp-platform/riscv-dbg/pull/129#discussion_r835654366).

File: hw/ip/rv_dm/rtl/rv_dm.sv
Comment (tjaychen) [inline hw/ip/rv_dm/rtl/rv_dm.sv:344]: does this comment need some updating now? we still don't generate alerts, but it looks like the error is now propagated.

**Review by tjaychen — APPROVED** [review]

**Review by sriyerg — APPROVED** [review]
LGTM, thanks!

File: hw/vendor/pulp_riscv_dbg/src/dm_top.sv
Comment (sriyerg) [inline hw/vendor/pulp_riscv_dbg/src/dm_top.sv:11]: I suppose you will instantiate this directly into RV_DM later?

Comment (msfschaffner) [reply hw/vendor/pulp_riscv_dbg/src/dm_top.sv:11]: No, the top files are replaced by `rv_dm` in OpenTitan.

The reason why I had to add them are the patches that are hosted temporarily on our lowRISC fork (until they get merged upstream).

The vendor-in tool would otherwise fail to apply these patches when we exclude the top-level files `dm_top.sv` and `dm_obi_top.sv` from being vendored in , since the patches contain changes in these two files.

I will remove these files with a fresh vendor-in once the patches have been merged upstream.

File: hw/ip/rv_dm/rtl/rv_dm.sv
Comment (msfschaffner) [reply hw/ip/rv_dm/rtl/rv_dm.sv:344]: Indeed, I have added a third point here explaining that the error is now latched in an error CSR that can be read out via JTAG.

File: hw/vendor/pulp_riscv_dbg/src/dm_top.sv
Comment (msfschaffner) [reply hw/vendor/pulp_riscv_dbg/src/dm_top.sv:11]: This is actually a good idea! 

It used to be that `dm_top` contained AXI-lite adapters, hence we had to replicate most of the logic here separately. 

But now it appears that the AXI bus adapters have been split out from `dm_top` into `dm_obi_top` upstream - so this constraint is no longer present.

I have restructured `rv_dm` a bit and instantiated `dm_top` directly.

**Comment by msfschaffner** [comment]
I just noticed that the patch got merged upstream: https://github.com/pulp-platform/riscv-dbg/pull/129#discussion_r835654366

I have hence re-vendored the `rv_dm` so that we do not have to rely on temp patches anymore.

**Review by sriyerg — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
